Updates to Smsdedbg per issue 21 and 26 (#35)
riscv/riscv-smmtt: chapter3.adoc

The `msdcfg` is a 32-bit read/write register, formatted as shown in <<MSDCFG>>.
This CSR is used by M-mode software to specify the active configuration for
capabilities of the supervisor domain when associated with a hart. +
The following extensions use the `msdcfg` register to specify additional
configuration for supervisor domains:

. `Smsdia` uses `msdcfg.SDICN` to specify the active configuration for
  the supervisor domain interrupt controller associated with the hart.

> . `Smsdedbg` specifies the `msdcfg.sdedbgalw` bit to manage external-debug for a supervisor domain.
> . `Smsdetrc` specifies the `msdcfg.sdetrcalw` bit to manage external-trace for a supervisor domain.

Author: ```suggestion
. `Smsdedbg` specifies the `msdcfg.sdedbgalw` bit to manage 
  external-debug for a supervisor domain.
. `Smsdetrc` specifies the `msdcfg.sdetrcalw` bit to manage 
  external-trace for a supervisor domain.
```

The `msdcfg` is a 32-bit read/write register, formatted as shown in <<MSDCFG>>.
This CSR is used by M-mode software to specify the active configuration for
capabilities of the supervisor domain when associated with a hart. +
The following extensions use the `msdcfg` register to specify additional
configuration for supervisor domains:

. `Smsdia` uses `msdcfg.SDICN` to specify the active configuration for
  the supervisor domain interrupt controller associated with the hart.

> . External-debug-allowed state for a supervisor domain is managed via the

Reviewer: The next two bullets would want to use similar text as first and last bullet in specifying the extension names and field.

Author: ack

> . External-debug-allowed state for a supervisor domain is managed via the
>   `msdcfg.sdedbgalw` bit.
> . External-trace-allowed state for a supervisor domain is managed via the
>   `msdcfg.sdetrcalw` bit.

Author: ```suggestion
. `Smsdedbg` specifies the `msdcfg.sdedbgalw` bit to manage external-debug for a supervisor domain.
. `Smsdetrc` specifies the `msdcfg.sdetrcalw` bit to manage external-trace for a supervisor domain.
```
